### PR TITLE
front: split infra editor files

### LINKED ISSUE #6029: front: refacto length of editor components (too much code in solo files)
The length of some files from infrastructure editor is very high (~500-1000 lines), the files should be split and reorganized with the new files.

### PR DESCRIPTION
**Pull request by clarani**
closes #6029

### CODE REVIEW

**Review by Akctarus — APPROVED** [review]
LGTM, tested with all the edition tools, it works well !

File: front/src/applications/editor/tools/routeEdition/components/index.tsx
Comment (SharglutDev) [inline front/src/applications/editor/tools/routeEdition/components/index.tsx:1]: As index.tsx is the kind of the view of routeEdition, should he still be put in the components folder ?

Also for harmonization, even if this file is already small, we should maybe extract the Messages component into a separate file. 

Actually I'm not even sure this whole index file is necessary

File: front/src/applications/editor/Editor.scss [hunk 1]
```diff
@@ -181,18 +181,33 @@
...
+      min-height: 100px;
+    }
   }
-  #errors-list-container {
-    height: 60vh;
-    overflow: auto;
+
+  .z-index-tooltip {
+    z-index: 1070 !important;
+  }
```
Comment (SharglutDev) [inline front/src/applications/editor/Editor.scss:202]: Not sure this needs to be in this css file as its linked to the navbar. But it doesn't concern this PR :)

**Review by SharglutDev — COMMENTED** [review]
Thank you for this PR and this great refacto. Everything works fine, left a few small suggestions

File: front/src/applications/editor/Editor.scss (hunk 1 above)
Comment (clarani) [reply front/src/applications/editor/Editor.scss:202]: Yes I agree :) there is already this refacto ticket https://github.com/osrd-project/osrd/issues/5421 about z index

**Review by SharglutDev — APPROVED** [review]
Lgtm !

**Pull request merged**